web/crux-ui/src/hooks/use-dyo-formik.ts

```diff
@@ -1,17 +1,39 @@
+import { yupErrorTranslate } from '@app/validations'
 import { FormikConfig, FormikValues, useFormik } from 'formik'
+import { Translate } from 'next-translate'
 import { MutableRefObject, useEffect } from 'react'
 
-type DyoFormikOptions<Values> = FormikConfig<Values> & {
+export type DyoFormikOptions<Values> = FormikConfig<Values> & {
   submitRef?: MutableRefObject<() => Promise<any>>
+  t?: Translate
 }
 
 const useDyoFormik = <Values extends FormikValues>(options: DyoFormikOptions<Values>) => {
-  const { submitRef, ...formikOptions } = options
+  const { submitRef, t, ...formikOptions } = options
 
   const formik = useFormik({
     validateOnBlur: false,
     validateOnChange: false,
     ...formikOptions,
+    validationSchema: options.validationSchema
+      ? () => ({
+          ...options.validationSchema,
+          validateSync: (values: any, validationOptions: any) => {
+            try {
+              return options.validationSchema.validateSync(values, validationOptions)
+            } catch (err) {
+              throw t ? yupErrorTranslate(err, t) : err
+            }
+          },
+          validate: (values: any, validationOptions: any) =>
```

Comment:
I believe you can get rid of the `new Promise` with declaring this as `async` and a `try catch`.